File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& a, int n) {
        int push {} , pop {} , cur {0};
        vector <string> ans;
        
        for (int i = 0;i < a.size(); i++){
            if (cur + 1 == a[i])
                ans.push_back("Push") , cur = a[i];
            else {
                int diff = a[i] - cur;
                
                while (diff > 1)
                    ans.push_back("Push") , ans.push_back("Pop") , diff --;
                
                ans.push_back("Push");
                cur = a[i];
            }
        }
        
        return ans;
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cs
public class Solution {
    public IList<string> BuildArray(int[] target, int n) {
        int cur = 0;
        
        IList <string> ans = new List<string> ();
        for (int i = 0;i < target?.Length; i++){
            if (cur + 1 == target[i]){
                cur = target[i];
                ans.Add ("Push");
            }
            else {
                int diff = target[i] - cur;
                
                while (diff > 1){
                    ans.Add("Push");
                    ans.Add("Pop");
                    diff --;
                }
                ans.Add("Push");
                cur = target[i];
            }
        }
        
        return ans;
    }
}